### PR TITLE
add urqlClient to Page Context

### LINKED ISSUE #14: Best Practice to access urqlClient in getInitialProps
I am wondering what would be the best practice to access the urqlClient in a getInitialProps? Without next-urql I inject the urqlClient into the ctx, but that doesn't seem to happen with next-urql

### PR DESCRIPTION
**Pull request by BjoernRave**
Not sure if this is the best way, feel free to change it if you want

Closes #14

### CODE REVIEW

File: src/with-urql-client.tsx [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
   dedupExchange,
-  cacheExchange,
-  fetchExchange,
   Exchange,
+  fetchExchange,
+  Provider,
 } from 'urql';
-import { SSRExchange, SSRData } from 'urql/dist/types/exchanges/ssr';
-
+import { SSRData, SSRExchange } from 'urql/dist/types/exchanges/ssr';
```
Comment (parkerziegler) [inline src/with-urql-client.tsx]: Do you have some sort of plugin enabled that sorts `import`s alphabetically? I'd prefer to remove these from the diff if you're ok with that as it's not meaningful to the changes being introduced.

Comment (BjoernRave) [reply src/with-urql-client.tsx]: upps, yeah vscode is doing that for me, will remove it

**Comment by BjoernRave** [comment]
Also is there maybe a way to inject the type information of urqlClient into the wrapped component?

**Comment by parkerziegler** [comment]
> Also is there maybe a way to inject the type information of urqlClient into the wrapped component?

I'll have to think about this. Do you mean that if I have a component, called `Page`, that inside of `Page`'s `getInitialProps` I should be able to safely access `ctx.urqlClient`? Or is this more about being able to access `urqlClient` via `props` when rendering `Page`, i.e.:

```typescript
const Page: React.FC<{ urqlClient: Client }> = ({ urqlClient }) => {
   // use urqlClient as you like
};
```

I'd expect both of the above to work right now, but I haven't yet tested it out myself.

Also, one last request for this PR – can you update the `index.d.ts` [here](https://github.com/FormidableLabs/next-urql/blob/master/index.d.ts#L22) to include `urqlClient: Client` (just as you did in the source file). Ideally we'll auto-generate this definition file in the future and it'll get run pre-commit or something like that. Thanks again for contributing!

**Comment by BjoernRave** [comment]
Yea I was talking about being able to safely access urqlClient in `ctx.urqlClient`.

> Also, one last request for this PR – can you update the index.d.ts here to include urqlClient: Client (just as you did in the source file). Ideally we'll auto-generate this definition file in the future and it'll get run pre-commit or something like that. Thanks again for contributing!

Will do that

File: index.d.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ interface WithUrqlInitialProps {
 
 interface PageProps {
   pageProps?: WithUrqlClient;
+  urqlClient: Client;
```
Comment (parkerziegler) [inline index.d.ts]: This actually needs to go on the `NextContextWithAppTree`. `PageProps` is a special interface we use for the `_app.js` case, where the component defined in `_app.js` literally receives a prop called `pageProps` that has all of the wrapped Page's props inside it. So just move this line into the `NextContextWithAppTree` and we should be good!

**Review by parkerziegler — APPROVED** [review]
Thanks again!

**Comment by BjoernRave** [comment]
@parkerziegler is it possible to do a release for this?

**Comment by parkerziegler** [comment]
Yes, sorry about this @BjoernRave, been caught up in a few other things. I should have a release up in the next half hour, thanks for your patience.

**Comment by parkerziegler** [comment]
Alright @BjoernRave , should be up on npm! https://github.com/FormidableLabs/next-urql/releases/tag/0.2.4

**Comment by BjoernRave** [comment]
@parkerziegler no problem, thanks alot :)

**Pull request merged**